wandb/sdk/internal/system/assets/interfaces.py

```diff
@@ -168,7 +168,13 @@ def start(self) -> None:
             logger.info(f"Started {self._process.name}")
 
     def finish(self) -> None:
-        if self._process is not None:
+        if self._process is None:
+            return None
+        if self._process.is_alive():
```

Comment:
im not so sure about this is_alive - join pairing.
I think the thing that saves you is only if shutdown_event is alway set in this case? is it?

otherwise your monitor loop might still be called on a thread that was just about to be started even thbough you just nulled out the process attribute.

Reply:
Removed the is_alive check, it's useless and won't fix anything. Replaced with a try/except: if the thread has not been started for some reason (in colab, we would stop the system metrics process with a pause request, so I could imagine something going wrong just before we do that while thing were still being set up), attempting to join it won't fail, but we'd still null the attribute.